### DISCUSSION TITLE
Consistent disconnect on fabric 1.20.1 local server

### DISCUSSION THREAD

**Discussion by seanlowe** [itzg/docker-minecraft-server#2499] [Q&A]
Hi, I've used this image across a variety of Forge servers and had minimal issues with them. I decided recently to try out a Fabric one and got it working with a little difficulty but I did get it working. 

**The Problem:** I will be playing on the server for like 5-10 minutes, and then it'll just kick me with a `internal exception: java.net.SocketException: connection reset` error. I go back to the server list, wait a little bit and can reconnect no issues.

Side note: I've noticed that if I have an open SSH session to the computer running the server, it will kick me from SSH at the same time I get disconnected from the server with the standard SSH error:
```bash
client_loop: send disconnect: Connection reset by peer
```

I'm running the server on a local machine I have at the house that I've cobbled together over the years after computer upgrades. 

Server info via screenfetch:
```
OS: Ubuntu 22.04 jammy
Kernel: x86_64 Linux 6.2.0-36-generic
Uptime: 7h 36m
Packages: 2066
Shell: bash 5.1.16
Disk: 37G / 458G (9%)
CPU: Intel Core i5-9600K @ 6x 4.6GHz [29.0°C]
GPU: UHD Graphics 630
RAM: 6541MiB / 64158MiB
```

Command used to create docker container:
```bash
docker run -d --name=prominence-2 \
 -e EULA=TRUE \
 -e MEMORY=4G \
 -e TYPE=CUSTOM \
 -e CUSTOM_SERVER=/data/fabric-server-launcher.jar \
 -e VERSION=1.20.1 \
 -p 25595:25565 \
 -v /home/minecraft/prominence-2:/data \
 itzg/minecraft-server:java20
```

Docker container notes:
As per [this link](https://github.com/itzg/docker-minecraft-server/discussions/1232), in order to create the container I unzipped the server pack into the directory I wanted, ran `start.sh` to have it create the server, and then ran the docker command how the link described before it would work properly.

Also, I've mapped the server's 25565 port to the host machine's 25595 port as I have multiple other minecraft docker containers with other modpacks

Modpack in question: Prominence II ([link](https://www.curseforge.com/minecraft/modpacks/prominence-2-fabric))

Client is running on Windows 11.

Server.properties:
```
#Minecraft server properties
#Sat Nov 18 17:40:43 UTC 2023
allow-flight=true
allow-nether=true
broadcast-console-to-ops=true
broadcast-rcon-to-ops=true
difficulty=easy
enable-command-block=true
enable-jmx-monitoring=false
enable-query=false
enable-rcon=true
enable-status=true
enforce-secure-profile=true
enforce-whitelist=true
entity-broadcast-range-percentage=100
force-gamemode=false
function-permission-level=2
gamemode=survival
generate-structures=true
generator-settings={}
hardcore=false
hide-online-players=false
initial-disabled-packs=data/SkyVillages-PracticalPlushiesCompat-1.20.1-1.0.3.2.zip,data/ichphilipp-s-endcity-better-end-1-18-1.zip,data/ElocinDev-Travelers-Comfort-Datapack.zip,data/SkyVillages-GraveyardCompat-1.20.1-1.0.3.zip,data/SkyVillages-WaystonesCompat-1.20.1-1.0.3-fabric.zip,data/Repurposed_Structures-Farmers_Delight_v6.zip,data/SkyVillages-SupplementariesCompat-1.20.1-1.0.3.1.zip,data/simplyswords_join_book.zip,data/WDA-VanillaLoot-1.18.2-1.19.2-0.0.1.zip,data/LessStructures-SpacingTweaks-1.20.1-2.1.56.zip,data/WDA-NoFlyingStructures-1.18.2-1.19.2.zip,data/Repurposed_Structures-Friends_And_Foes_v5.zip,data/SkyVillages-FarmersDelightCompat-1.20.1-1.0.3.zip
initial-enabled-packs=vanilla,fabric
level-name=world
level-seed=
level-type=default
max-build-height=256
max-chained-neighbor-updates=1000000
max-players=10
max-tick-time=120000
max-world-size=29999984
motd=A Minecraft Server
network-compression-threshold=256
online-mode=true
op-permission-level=4
player-idle-timeout=0
prevent-proxy-connections=false
pvp=true
query.port=25565
rate-limit=0
rcon.password=58849873f692d4db04ec118c
rcon.port=25575
require-resource-pack=false
resource-pack=
resource-pack-prompt=
resource-pack-sha1=
server-ip=
server-port=25565
simulation-distance=10
snooper-enabled=false
spawn-animals=true
spawn-monsters=true
spawn-npcs=true
spawn-protection=16
sync-chunk-writes=true
text-filtering-config=
use-native-transport=true
view-distance=10
white-list=false
```

Also, (just thought of this) whenever I reconnect after getting kicked, it will display a little popup saying "new config generated" or some such. Not sure if that's important.
If there's any other info I can provide, let me know. Thanks!

**Comment by itzg** [comment]
You're using quite a few custom things there, so it'll be tough for anyone to recreate the problem. In any case, this is a better question for the Fabric forums since I'm not able to support Minecraft server issues beyond the automated setup the image does.
